### PR TITLE
Message when trying to load modded profiles

### PR DESCRIPTION
**Pull request by ArgTang**
Hi

This will let users know that amidst need a vanilla profile, and returns to the profileselector afterwards, when trying to select a modded profile.

This is a small fix, and will probably not catch all modded profiles, but most of them. 
To get a better fix, we would have to check each profile with the profile list that we get from mojang, And mark all profiles that do not mach as modded and display some sort of message. I am not shure where or how to do this, but i can look into it if need be.

Let me know what you think.

### CODE REVIEW

**Comment by ArgTang** [comment]
#205

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = tab
+indent_size = 4
```
Comment (stefandollase) [inline .editorconfig]: Please don't make changes that are unrelated to the feature.

File: .gitignore [hunk 2]
```diff
@@ -5,9 +5,11 @@
...
 
 # IntelliJ IDEA project files #
-./AMIDST.eml
-./AMIDST.iml
-./.idea/*
+*.iml
+/.idea/
+/out/
+
+# Netbeans files
```
Comment (stefandollase) [inline .gitignore]: Please don't make changes that are unrelated to the feature.

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java [hunk 3]
```diff
@@ -1,7 +1,5 @@
 package amidst.gui.profileselect;
 
-import java.io.FileNotFoundException;
```
Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: Please use Eclipse to prevent changing the order of import statements.

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java [hunk 4]
```diff
@@ -81,6 +82,13 @@ private boolean tryLoad() {
 		try {
 			Log.i("using minecraft launcher profile '" + getProfileName() + "' with versionId '" + getVersionName()
 					+ "'");
+					
+			String possibleModProfiles = "optifine|forge";
+            if (Pattern.matches(possibleModProfiles, getProfileName().toLowerCase())) {
```
Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: `Pattern#matches` requires the complete profile name to match the pattern. Thus, this only works if the complete profile name is either the word "optifine" or "forge", without anything else. For example "OptiFine 1.8.9" does not work. My suggestion is to simply use a list of strings and check if the profile name `contains` any of the strings.

`String#toLowerCase` depends on the default `Locale` on the users computer. This is an issue, see [here](https://github.com/toolbox4minecraft/amidst/issues/151#issuecomment-211903792) and [here](http://stackoverflow.com/questions/11063102/using-locales-with-javas-tolowercase-and-touppercase). However, it should work to use `.toLowerCase(Locale.ENGLISH)`.

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java [hunk 5]
```diff
@@ -81,6 +82,13 @@ private boolean tryLoad() {
...
+                Log.e("Amidst does not support modded Profiles, please select or create an unmodded profile");
```
Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: I propose the error message: "Amidst does not support modded Minecraft profiles! Please select or create an unmodded Minecraft profile via the Minecraft Launcher."

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java [hunk 6]
```diff
@@ -81,6 +82,13 @@ private boolean tryLoad() {
...
+                return false;
+            }
```
Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: The if-block uses spaces instead of tabs for its indentation. Please use Eclipse and import the preference file that is included in this repository to ensure that your commits adhere to our coding style.

**Comment by stefandollase** [comment]
Sorry for the delay. There was no evil intent, I just forgot about it after shortly looking into it when you posted it.

This should be a quick fix for the problem. Thanks! Nevertheless, there still need to be some changes until this can be merged. I marked them in the review.

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java (hunk 4 above)
Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: One more thing: Please don't check against the profile name, but against the version name. The profile name is chosen by the user and can thus be something like "1.8.9-without-optifine". The current approach would block this profile. Using the version name will prevent this issue.

**Comment by stefandollase** [comment]
Also, please add the following comment in front of the introduced check:

```
// TODO: Replace with proper handling for modded profiles.
```

**Comment by ekevoo** [comment]
Requiring a specific IDE is a serious deterrent to get further contributors to any Open Source project. I was considering brushing my decades-outdated knowledge of Java and maybe start contributing, but not being able to use my IDE of choice (which most likely would be IntelliJ Idea since I'm a very satisfied IntelliJ PyCharm user) is a deal-breaker.

File: .gitignore (hunk 2 above)
Comment (ArgTang) [inline .gitignore]: Ill make a separate pull request for this

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java (hunk 3 above)
Comment (ArgTang) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: This is IDEA's auto reordering. ill try to manually move it back.

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java (hunk 4 above)
Comment (ArgTang) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: Yes that is badly tested by me, ill fix the regex
I was not aware of the locale.

**Comment by ArgTang** [comment]
@stefandollase This should adress the issues you brought up

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java (hunk 3 above)
Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: Yes. IDEA and Eclipse both have the *organize imports* feature, however they seem to choose a different ordering. This is why I prefer to have all developers use a single set of tools. This is just an explanation, I saw you already fixed it for this pull request.

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java [hunk 7]
```diff
@@ -78,9 +80,16 @@ public void load() {
...
 	@CalledOnlyBy(AmidstThread.WORKER)
 	private boolean tryLoad() {
+		// TODO: Replace with proper handling for modded profiles.
 		try {
-			Log.i("using minecraft launcher profile '" + getProfileName() + "' with versionId '" + getVersionName()
-					+ "'");
+			Log.i("using minecraft launcher profile '" + getProfileName() + "' with versionId '" + getVersionName() + "'");
+
+			String possibleModProfiles = ".*[optifine|forge].*";
+			if (Pattern.matches(possibleModProfiles, getVersionName().toLowerCase(Locale.ENGLISH))) {
```
Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: This is still broken. `[]` is used to represent a set of characters. For example `[hello]` means *a single character from the set {'h', 'e', 'l', 'o'}*.

Please use the following statement:

```
Pattern.compile("optifine|forge").matcher(getVersionName().toLowerCase(Locale.ENGLISH)).find()
```

If you want to, you can extract `Pattern.compile("optifine|forge")` as a constant.

**Comment by stefandollase** [comment]
@ArgTang
This is already better, however still not perfect. I added a comment.

@ekevoo @ArgTang
I see your point. However, as seen in this pull request, there are also good reasons to enforce a single set of tools for a project. Different tools often don't work that well together. However, lets not spam this pull request with a detailed discussion about it. If you want to further discuss this and want to try to change my opinion, please create a new discussion issue.

File: src/main/java/amidst/gui/profileselect/LocalProfileComponent.java (hunk 7 above)
Comment (ArgTang) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: wow, yeah. Thanks for your patience. I changed the regex again.

Comment (stefandollase) [inline src/main/java/amidst/gui/profileselect/LocalProfileComponent.java]: Great, this should also work.
